src/basic/os-util.c

```diff
@@ -317,3 +317,35 @@ int load_extension_release_pairs(const char *root, const char *extension, char *
 
         return load_env_file_pairs(f, p, ret);
 }
+
+int os_release_support_ended(const char *support_end, bool quiet) {
+        _cleanup_free_ char *t = NULL;
+        int r;
+
+        if (!support_end) {
+                r = parse_os_release(NULL,
+                                     "SUPPORT_END", &t);
+                if (r < 0)
+                        return log_full_errno(r == -ENOENT || quiet ? LOG_DEBUG : LOG_WARNING, r,
```

Comment:
So you're silently treating `true` as a `LOG_` constant? This looks ugly; is this used anywhere else?

Reply:
nvm, I read this as `r == -ENOENT || (quiet ? LOG_DEBUG : LOG_WARNING)`. Could use some `()` around the condition if you ask me.